### PR TITLE
[*/2017b] fix missing extensions in cutadapt v1.16

### PR DESCRIPTION
**Pull request by vanzod**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.8, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/5d3296dfe11b83a15b23ddcf8a3716bb for a full test report.

**Comment by boegel** [comment]
@vanzod How come this went undetected?

Also, why not apply the same fix to the `2018a` and `2018b` easyconfigs for `cutadapt` 1.16 & 1.18?

**Comment by vanzod** [comment]
@boegel I saw the problem when running the `cutadapt` executable. Unfortunately running `cutadapt -h` would work even with the missing packages, so it is not a reliable test anyway.
Since I have been working with the 2017b toolchain, I have not checked `cutadapt` from other toolchains. I'll take care of it.

**Comment by vanzod** [comment]
@boegel After a quick test of all versions of `cutadapt` provided in the common toolchains starting from 2016b, only the ones fixed in this PR for the 2017b toolchain are affected by the missing packages.

**Comment by boegel** [comment]
@vanzod How did you test exactly? Just rebuild the easyconfigs? Note that many of them don't include `download_dep_fail` or use `PythonBundle`...

File: easybuild/easyconfigs/c/cutadapt/cutadapt-1.16-foss-2017b-Python-3.6.3.eb [hunk 1]
```diff
@@ -19,16 +19,26 @@ description = """Cutadapt finds and removes adapter sequences, primers, poly-A t
...
-
 dependencies = [
     ('Python', '3.6.3'),
 ]
 
 use_pip = True
-download_dep_fail = True
+
+exts_list = [
+    ('bz2file', '0.98', {
```
Comment (boegel) [inline easybuild/easyconfigs/c/cutadapt/cutadapt-1.16-foss-2017b-Python-3.6.3.eb]: @vanzod I think this one is only needed for Python 2.7.x (see also last commit in #7562)

**Comment by vanzod** [comment]
I confirm that with `download_dep_fail = True` all other easyconfig files will need to be implemented as `PythonBundle` to satisfy the dependencies. I will do that in a separate PR.

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/34af263b43a196aed4bc812fd93c60bf for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3118.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/1a21b747e6223f1d1d46b099eca890a2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2406.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/862465de4eaf12acce90c893c364f1b0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2039.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/946a4015cc5c61463ceb49d362dce2ef for a full test report.

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**